pkg/controller/migmigration/validation.go

```diff
@@ -170,7 +170,8 @@ func (r ReconcileMigMigration) validateFinalMigration(plan *migapi.MigPlan, migr
 			}
 		}
 	}
-	if hasCondition {
+	// Allow perform Rollback on finished Plan.
+	if hasCondition && !migration.Spec.Rollback {
```

Comment:
@djwhatle I'm not sure this is the right approach, but in order to allow run Rollback on completed migration, I needed to modify hasFinalMigration validation.